src/type-layout.md

```diff
@@ -184,7 +184,10 @@ for field in struct.fields_in_declaration_order() {
     // Increase the current offset so that it's a multiple of the alignment
     // of this field. For the first field, this will always be zero.
     // The skipped bytes are called padding bytes.
-    current_offset += field.alignment % current_offset;
```

Comment:
There's another "current_offset + current_offset % struct.alignment" further down, why doesn't that need fixing, too?